File: xv/XV5080FX.swift

extension XV5080 {
  
  enum FX {
    static let patchWerk = XV.FX.patchWerk(params: parms.params())

    static let fxTypeOptions: [Int : String] = XV5050.FX.fxTypeOptions
    
    static let parms = XV5050.FX.parms + [
      .p([.out], 0x04, .options(outOptions)),
    ]
    
    static let outOptions: [Int:String] = OptionsParam.makeOptions(["A","B","C","D"])
  }

  enum Chorus {
    static let patchWerk = XV.Chorus.patchWerk(params: parms.params())

    static let parms = XV5050.Chorus.parms + [
      .p([.out, .assign], 0x02, .options(FX.outOptions)),
    ]
  }

  enum Reverb {
    static let patchWerk = XV.Reverb.patchWerk(params: parms.params())

    static let parms = XV5050.Reverb.parms + [
      .p([.out, .assign], 0x02, .options(FX.outOptions)),
    ]
  }
}

File: xv/xv5080_fx.js
const XVFX = require('./xv_fx.js')
const XV5050FX = require('./xv5050_fx.js')

const fxTypeOptions = XV5050FX.fxTypeOptions

const outOptions = ["A","B","C","D"]

const parms = XV5050FX.parms.concat([
  ['out', { b: 0x04, opts: outOptions }],
])

const chorusParms = XV5050FX.chorusParms.concat([
  ['out/assign', { b: 0x02, opts: outOptions }],
])

const reverbParms = XV5050FX.reverbParms.concat([
  ['out/assign', { b: 0x02, opts: outOptions }],
])

module.exports = {
  patchWerk: XVFX.patchWerk(parms),
  chorusPatchWerk: XVFX.chorusPatchWerk(chorusParms),
  reverbPatchWerk: XVFX.reverbPatchWerk(reverbParms),
}


